src/lightning/fabric/utilities/load.py

```diff
@@ -0,0 +1,203 @@
+        from torch.storage import TypedStorage, UntypedStorage
+
+        name, storage_cls, fn, device, size = self.storageinfo
+        dtype = self.metatensor.dtype
+
+        storage = self.archiveinfo.file_reader.get_storage_from_record(
+            f"data/{fn}", size * torch._utils._element_size(dtype), UntypedStorage
+        )
+        uts = storage._typed_storage()._untyped_storage
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            storage = TypedStorage(wrap_storage=uts, dtype=dtype, _internal=True)
+        return torch._utils._rebuild_tensor_v2(storage, *self.rebuild_args)
+
+    @classmethod
+    def __torch_function__(
+        cls,
+        func: Callable,
+        types: Sequence,
+        args: Sequence[Any] = (),
+        kwargs: Optional[Dict] = None,
+    ) -> Any:
+        kwargs = kwargs or {}
+        loaded_args = [(arg._load_tensor() if isinstance(arg, _NotYetLoadedTensor) else arg) for arg in args]
+        return func(*loaded_args, **kwargs)
+
+    def __getattr__(self, name: str) -> Any:
+        # These properties don't require materialization and can be accessed through the meta tensor directly
+        if name in {
+            "dtype",
+            "grad",
+            "grad_fn",
+            "layout",
+            "names",
+            "ndim",
+            "output_nr",
+            "requires_grad",
+            "retains_grad",
+            "size",
+            "shape",
+            "volatile",
+        }:
+            return getattr(self.metatensor, name)
+
+        # TODO: needed for us?
+        # materializing with contiguous is needed for quantization
+        if name in {"contiguous"}:
+            return getattr(self._load_tensor(), name)
+
+        raise AttributeError(f"{type(self)} does not have {name}")
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({repr(self.metatensor)})"
+
+
+class _LazyLoadingUnpickler(pickle.Unpickler):
+    def __init__(self, file: IO, file_reader: torch.PyTorchFileReader) -> None:
+        super().__init__(file)
+        self.file_reader = file_reader
+
+    def find_class(self, module: str, name: str) -> Any:
+        if module == "torch._utils" and name == "_rebuild_tensor_v2":
+            return partial(_NotYetLoadedTensor.rebuild_tensor_v2, archiveinfo=self)
+        if module == "torch._tensor" and name == "_rebuild_from_type_v2":
+            return partial(_NotYetLoadedTensor.rebuild_from_type_v2, archiveinfo=self)
+        if module == "torch._utils" and name == "_rebuild_parameter":
+            return partial(_NotYetLoadedTensor.rebuild_parameter, archiveinfo=self)
+        return super().find_class(module, name)
+
+    def persistent_load(self, pid: tuple) -> "TypedStorage":
+        from torch.storage import TypedStorage
+
+        name, cls, fn, device, size = pid
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")  # TODO: needed?
```

Comment:
The TypedStorage APIs have heavy deprecations in torch. This is okay with me since it's wrapped by `catch_warnings`

Reply:
After your comment, I added a comment to the code that this is the reason why we suppress warnings